eclipse/ice
# Create top-level POM for ICE + BATS refactoring to accomodate

Fixes #488 

*Work includes:*
 - Moving the `org.eclipse.ice.build` pom.xml to the top-level of the project
 - Removing the `org.eclipse.ice.build` package entirely
 - Modifying BATS so you no longer need to have the `docker:build` if you do `mvn clean docker:build verify` (now just need `mvn clean verify`
 - Removed BATS sub-module from the top-level pom.xml `default` profile
 - Added a `full` profile to top-level pom.xml that includes building ICE with BATS (requires Docker)

*To test:*
1) Ensure can build ICE at the top-level using `mvn clean verify` and/or `mvn clean install`
2) When doing this, ensure BATS is NOT built
3) Now, with Docker installed, go to `org.eclipse.ice.bats` module and run `mvn clean verify`. Should see that a Fuseki docker image is built in the `pre-integration-test` phase and the integration tests run.
4) Go back to the top-level of ICE and now run the `full` profile so we including building BATS. Run `mvn clean verify -P full` and/or `mvn clean install -P full`. Ensure BATS is built.
